extensions/2.0/Vendor/EXT_lights_environment/README.md

```diff
@@ -0,0 +1,128 @@
+# EXT_lights_environment
+
+## Contributors
+
+* `TODO: Add contributors`
+
+## Status
+
+Draft
+
+## Dependencies
+
+Written against the glTF 2.0 spec.
+
+## Overview
+
+This extension provides the ability to define an environment light (also known as [sky light](https://docs.unrealengine.com/en-US/Engine/Rendering/LightingAndShadows/LightTypes/SkyLight/index.html)) in a glTF scene. In general, this environment is used for image based lighting in rasterizers and pathtracers.  
+
+To environment light is defined by a panorama image (also know as [equirectangular image](https://docs.blender.org/manual/en/latest/render/lights/world.html)). For encoding the panorama image and for supporting [High-dynamic-range rendering](https://en.wikipedia.org/wiki/High-dynamic-range_rendering), the widely used [RGBE image format](https://en.wikipedia.org/wiki/RGBE_image_format) is used as a `.hdr` file:  
+The panorama image and `.hdr` file format is supported in paint tools like Gimp or Photoshop, 3D modelling tools like Blender or 3ds Max, Game Engines like Unity or Unreal Engine and Web Engines like Babylon.js or three.js.  
+In general, this format is already used to define environment lights in todays 3D engines.  
+
+How to pre-filter and/or sample the panorma image is implicitly definded by the [materials BRDF](https://github.com/KhronosGroup/glTF/tree/master/specification/2.0#appendix-b-brdf-implementation). 
+
+### Why no pre-filtering (also known as pre-sampling)?
+
+* Pathtracers do not require this extra data
+* Rasterizers can decide between Spherical Harmonics vs. pre-filtered diffuse lighting
+* Rasterizers can decide ...
+  * Numbers of pre-filtering samples
+  * Pre-filtering algorithm and optimizations
+  * Direct panorama image sampling vs. cube map sampling  
+    * Cube map resolution
+    * Cube map mip-map optimization
+* Pre-filtering generates more data to be transfered  
+  * Each BRDF potentially needs a new look up table and a mip-mapped cube map
+* Graphics API independent
+  * DirectX vs. OpenGL 
+
+## Defining an environment light
+
+The `EXT_lights_environment` extension requires to extend the following glTF schemas.
+
+### Defining the panorama image
+
+```json
+"images": [
+  {
+    "uri": "any_other_image.png"
+  },
+  {
+    "uri": "dodge2.hdr"
+  }
+]
+```
+
+This means, that the `mimeType` inside [image.schema.json](https://github.com/KhronosGroup/glTF/blob/master/specification/2.0/schema/image.schema.json) has to support `image/vnd.radiance`.
+
+### Defining the environment light
+
+One or more environment lights cane be defined inside [glTF.schema.json](https://github.com/KhronosGroup/glTF/blob/master/specification/2.0/schema/glTF.schema.json)
+
+```json
+"extensions": {
+  "EXT_lights_environment": {
+    "environments": [
+      {
+        "source": 0,
+        "intensity": 1.0,
+        "frontside": "+X"
```

Comment:
Is it necessary that we allow this to be configured, rather than normatively stating a front facing direction as we do with cameras?

Reply:
The problem is, that different DCC tools do have a different front face. Some do have +Z, some do have +X. So, yes this is required to compensate this.

Reply:
Some engines will have different conventions too — we are placing the burden of conversion on them, rather than the DCC tools? This is much easier with cubemaps, perhaps.

Reply:
E.g. the above screenshot from Blender does have +X as the front face. Because of Blender's coordinate system.
If you set up the same scene in 3ds max - as there +Z is the front face, the scene looks different. To compensate, during export to glTF, the front face is defined.
If an engine is following the glTF specification, the coordinate system is well defined and I do not see any burden. It is the opposite, now the orientation is well defined.
Also, some 3D engines will not use cube maps. E.g. Blender Cycles can be executed on the CPU and there is no need to convert to a cube map. And probably any CPU based path/raytracer will directly sample from the panorma image.